Meme-ify-Me/app/src/main/java/nyc/c4q/rosmaryfc/meme_ify_me/MainActivity.java

```diff
@@ -95,6 +91,40 @@ private void takePhoto (View v){
         startActivityForResult(intent, TAKE_PICTURE);
     }
 
+    //method for gathering intent information from takePhoto and pickPhoto methods
+    // and setting the imageview with correct bitmap
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data){
+        if(resultCode == RESULT_OK ){
+                if (requestCode == PICK_PICTURE){
+                    selectedImagePath = String.valueOf(data.getData());
+                    imageview.setImageBitmap(decodePhoto(selectedImagePath));
+                } else if (requestCode == TAKE_PICTURE){
+                    selectedImagePath = imageUri.toString();
+                    imageview.setImageBitmap(decodePhoto(selectedImagePath));
+                } else {
+                    super.onActivityResult(requestCode, resultCode, data);
+                }
+        }
+    }
+
+    //requesting image's file path and converting into url and calling the
+    // ContentResolver to retrieve image and set it inside a bitmap
+    public Bitmap decodePhoto (String path) {
+        Uri selectedImageUri = Uri.parse(selectedImagePath);
```

Comment:
We need to make URl selectedImageUri as global so that I can call it in my new method.